gap/base/methods.gd

```diff
@@ -0,0 +1,130 @@
+BindGlobal("RecogMethodType",
+           NewType(RecogMethodFamily,
+                   IsRecogMethod and IsAttributeStoringRep));
+
+# <#GAPDoc Label="RecogMethod">
+# <ManSection>
+# <Func Name="RecogMethod" Arg="stamp, comment, func"/>
+# <Description>
+# Return a recognition method <C>method</C> in the filter
+# <Ref Filt="IsRecogMethod"/>,
+# where <A>stamp</A> is a string describing the method uniquely,
+# <A>comment</A> is a string explaining how the method works, and
+# <A>func</A> is the method itself.
+# These components of <C>method</C> can be accessed via the attributes
+# <Ref Attr="Stamp"/>, <Ref Attr="Comment"/> and the function
+# <Ref Func="UnpackRecogMethod"/>, respectively.
+# <P/>
+# The recognition method <C>method</C> can be called as if it were equal to the
+# function <C>func</C>: namely <C>method(args)</C> delegates directly to
+# <C>func(args)</C>.
+# <P/>
+# A recognition method returns one of the following four values:
+# <List>
+# <Mark><K>Success</K></Mark>
+# <Item>means that the method was successful and no more methods have to be
+#     tried.</Item>
+# <Mark><K>NeverApplicable</K></Mark>
+# <Item>means that the method was not successful and that there is no point
+#     to call the method again in this situation whatsoever.</Item>
+# <Mark><K>TemporaryFailure</K></Mark>
+# <Item>means that the method temporarily failed, that it however could be
+#     sensible to call it again in this situation at a later stage. This
+#     value is typical for a Las Vegas algorithm using randomised methods,
+#     which has failed, but which may succeed when called again.</Item>
+# <Mark><K>NotEnoughInformation</K></Mark>
+# <Item>means that the method for some reason refused to do its work. However,
+#     it is possible that it will become applicable later such that it makes
+#     sense to call it again, for example when more information is available.</Item>
+# </List>
+# </Description>
+# </ManSection>
+# <#/GAPDoc>
+DeclareGlobalFunction("RecogMethod");
+
+# <#GAPDoc Label="BindRecogMethod">
+# <ManSection>
+# <Func Name="BindRecogMethod" Arg="r, arg"/>
+# <Description>
+# Create the recognition method <C>method</C> by calling
+# <Ref Func="RecogMethod"/> with arguments <A>arg</A>.
+# Then bind the component <C>Stamp(method)</C> of <A>r</A> to <A>method</A>.
+# </Description>
+# </ManSection>
+# <#/GAPDoc>
+DeclareGlobalFunction("BindRecogMethod");
+
+# <#GAPDoc Label="UnpackRecogMethod">
+# <ManSection>
+# <Func Name="UnpackRecogMethod" Arg="method"/>
+# <Description>
+# Return the function underlying <A>method</A>.
+# The argument <A>m</A> must lie in <Ref Filt="IsRecogMethod"/>.
+# </Description>
+# </ManSection>
+# <#/GAPDoc>
+DeclareGlobalFunction("UnpackRecogMethod");
+
+# <#GAPDoc Label="CallRecogMethod">
+# <ManSection>
+# <Func Name="CallRecogMethod" Arg="m, args"/>
+# <Description>
+# Call <C>UnpackRecogMethod(m)</C> with arguments <A>args</A> and return the
```

Comment:
Maybe add a comment here that `m(args)` does essentially the same.

Reply:
After reading the review of Max, probably it would be better to drop `m(args)` entirely.

Reply:
I've removed the `m(args)` functionality as @fingolfin suggested.